File: frontend/types/analytics.ts
import type { Grievance } from '@/types/grievance';

export interface KpiCards {
  total: number;
  open: number;
  critical: number;
  avg_score: number;
}

export interface CategoryCount {
  category: string;
  count: number;
}

export interface WardCount {
  ward_id: number;
  ward_name: string;
  count: number;
}

export interface TrendPoint {
  date: string;
  count: number;
}

export interface AnalyticsSummary {
  kpis: KpiCards;
  categories: CategoryCount[];
  wards: WardCount[];
  trends: TrendPoint[];
}

File: backend/app/schemas/analytics.py
from datetime import date

from pydantic import BaseModel


class KpiCards(BaseModel):
    total: int
    open: int
    critical: int
    avg_score: float


class CategoryCount(BaseModel):
    category: str
    count: int


class WardCount(BaseModel):
    ward_id: int
    ward_name: str
    count: int


class TrendPoint(BaseModel):
    date: date
    count: int


class AnalyticsSummary(BaseModel):
    kpis: KpiCards
    categories: list[CategoryCount]
    wards: list[WardCount]
    trends: list[TrendPoint]
